linear-base.cabal

```diff
@@ -175,38 +175,21 @@ benchmark mutable-data
   ghc-options: -rtsopts=ignore
   default-language: Haskell2010
 
--- TODO: Uncomment below block and set 'build-type' to 'Custom' to enable
--- doctests once cabal-install 3.4 is released.
---
--- Longer story:
---
--- cabal-install has a piece of code[1] which injects a Cabal upper bound to
--- packages with custom Setup.hs's. And this happens after the overrides,
--- so the usual mechanisms of overriding upper bounds does not work.
---
--- GHC 9 comes with Cabal 3.4, which is above that bound. So, when using
--- GHC 9 with cabal-install 3.2; `build-type: Custom` causes another Cabal
--- library to be built, and that causes a strange type error ("expecting IO,
--- but got IO"), which I suspect because it conflicts with the existing boot
--- packages.
---
--- [1]: https://github.com/haskell/cabal/blob/d28c80acc69b9e7fa992a0b2b7fced937734b238/cabal-install/src/Distribution/Client/ProjectPlanning.hs#L1132-L1149
+custom-setup
+ setup-depends:
+   base >= 4 && <5,
+   Cabal,
+   cabal-doctest
```

Comment:
It would be nice having cabal-doctest behind a flag, as packages using `linear-base` don't need doctest to build the dependency.

Reply:
AFAIK using `cabal-docspec` doesn't require anything in the cabal file. So this particular aspect would be solved.